### PR TITLE
Themes, added `apply` for `MuiSpacings`, `copy` for `MuiSpacings` and…

### PR DESCRIPTION
**Pull request by oyvindberg**
… `MuiPalette`

### CODE REVIEW

**Comment by oyvindberg** [comment]
I made these as i was playing a bit with themes. we should really complete the support and write a demo or two for it. interested, @codingismy11to7 ?

File: core/src/main/scala/chandu0101/scalajs/react/components/materialui/MuiStyles.scala [hunk 1]
```diff
@@ -58,6 +114,34 @@ object MuiPalette {
       alternateTextColor = alternateTextColor, canvasColor = canvasColor, borderColor = borderColor,
       disabledColor = disabledColor).asInstanceOf[MuiPalette]
   }
+
+  implicit class MuiPaletteOpt(p: MuiPalette) {
```
Comment (codingismy11to7) [inline core/src/main/scala/chandu0101/scalajs/react/components/materialui/MuiStyles.scala]: oops, should be `Ops` i assume

Comment (oyvindberg) [inline core/src/main/scala/chandu0101/scalajs/react/components/materialui/MuiStyles.scala]: indeed! :)

**Comment by codingismy11to7** [comment]
i'm interested, yeah, but we're hard at work on bringing up a sjs-react UI, and things are just busy this time of year, so I don't know if i'll be able to contribute anything. I'll be around, (and I just posted the materialui icons project), so maybe if i get a minute i can help out with some of those sundries. anyway, thanks for merging, i'll try to help if i have time, but i'm pretty swamped these days. good job that you guys are doing.

**Pull request merged**